### PR TITLE
NP-884 Added publication fields doi and doiRequest

### PR DESCRIPTION
**Pull request by nraanes**

### CODE REVIEW

File: src/main/resources/publicationFrame.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "isOpenAccess": "openAccess",
     "isPeerReviewed": "peerReviewed",
     "hasDoi": "doi",
+    "hasDoiRequest": "doiRequest",
```
Comment (nraanes) [inline src/main/resources/publicationFrame.json:49]: I was not sure about adding this line, but it seemed appropriate.

File: src/main/java/no/unit/nva/model/DoiRequest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.Objects;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+public class DoiRequest {
+
+    private DoiRequestStatus status;
+    private Instant date;
+
+    public DoiRequest() {
+
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/DoiRequest.java:16]: space

File: src/main/java/no/unit/nva/model/DoiRequestStatus.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package no.unit.nva.model;
+
+public enum DoiRequestStatus {
+    REQUESTED,
+    ACCEPTED,
```
Comment (brinxmat) [inline src/main/java/no/unit/nva/model/DoiRequestStatus.java:5]: APPROVED?

File: src/main/resources/publicationFrame.json (hunk 1 above)
Comment (brinxmat) [reply src/main/resources/publicationFrame.json:49]: Yes, it is for the mapping to the external model

**Pull request merged**